File: src/concatenate_prime.java
import java.util.*;

public class concatenate_prime {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        HashMap<Long, Long> prime = new HashMap<Long, Long>();
        HashMap<Long, Long> store = new HashMap<Long, Long>();
        long num = sc.nextLong();
        long last = 0, temp = 3, length, count = 0,ncount = 0;
        boolean flag = false;
        prime.put((long) 0, (long) 2);
        while (temp <= num && num > 3) {
            count = 0;
            flag = false;
            length = prime.size();

            while (count < length) {
                long get = prime.get(count);
                if (get * get > temp) {
                    count=length;
                    break;
                } else if (temp % get == 0) {
                    flag = true;
                    break;
                }
                count++;
            }
            if (flag == false) {
                prime.put(count, temp);
                last= temp;
            }
            temp += 2;
        }
        int fc=prime.size();
        while (temp <= last*(long)Math.pow(10,(""+last).length())+last && num > 3) {
            count = 0;
            flag = false;
            length = prime.size();

            while (count < length) {
                long get = prime.get(count);
                if (get * get > temp) {
                    count=length;
                    break;
                } else if (temp % get == 0) {
                    flag = true;
                    break;
                }
                count++;
            }
            if (flag == false) {
                prime.put(count, temp);
            }
            temp += 2;
        }
        for (long i = 0; i < fc; i++)
            for (long j = 0; j < fc; j++)
            {
                long t=prime.get(i)*(long)Math.pow(10,(""+prime.get(j)).length())+prime.get(j);
                if(!store.containsValue(t)&&prime.containsValue(t))
                    store.put(ncount++,t);
            }
        System.out.println(ncount);
    }
}

File: src/concatenate_prime.py
n=input("");
def concatenate(n):
    c=0;
    list=[];
    a=[];
    arr=[];
    store=[];
    k=0;
    last=0;
    for i in range(2,n+1):
        c=0;
        for j in range(1,i+1):
            if i%j==0:
                c=c+1;
        if c==2:
            list.append(i);
            store.append(i);
            last=i;
    for i in range(n+1,int(str(last)+str(last))):
        c=0;
        for j in range(1,i+1):
            if i%j==0:
                c=c+1;
        if c==2:
            list.append(i);
    for i in store:
        for j in store:
            if int(str(i)+str(j))in list:
                if int(str(i)+str(j)) not in a:
                    a.append(int(str(i)+str(j)))
    print len(a)


concatenate(n);
